### ISSUE TITLE
'write_to' configuration parameter does not work

### ISSUE THREAD

**Issue by mr-rodgers** [pypa/setuptools-scm#33]
I'm using setuptools 18.0.1 with the following setup script:

``` python
setup(
    ...
    setup_requires=['setuptools_scm'],
    use_scm_version={'write_to': 'package/VERSION'},
    ...
)
```

Running the script like this just uses the default version that I gave to the setup script. However, if I change the parameter to `use_scm_version=True`, then it all works fine. The README says that I should be able to use the mapping as shown above, so this looks like a bug.

**Comment by RonnyPfannschmidt** [comment]
can you clarify what you mean with the 'default version',

also please try the master branch o setuptools_scm, it errors visibly in more cases

**Comment by mr-rodgers** [comment]
Trying the code from the master branch of setuptools_scm, I don't get an error message.

``` shell
(testenv) C:\Dev\findig [develop +0 ~3 -0]> python setup.py bdist_wheel
running bdist_wheel
running build
running build_py
installing to build\bdist.win32\wheel
running install
running install_lib
running install_egg_info
running egg_info
writing top-level names to Findig.egg-info\top_level.txt
writing Findig.egg-info\PKG-INFO
writing requirements to Findig.egg-info\requires.txt
writing dependency_links to Findig.egg-info\dependency_links.txt
reading manifest template 'MANIFEST.in'
writing manifest file 'Findig.egg-info\SOURCES.txt'
Copying Findig.egg-info to build\bdist.win32\wheel\Findig-0.1.0.data\..\Findig-0.1.0-py3.4.egg-info
running install_scripts
creating build\bdist.win32\wheel\Findig-0.1.0.dist-info\WHEEL
(testenv) C:\Dev\findig [develop +0 ~3 -0]>
```

The 'default version' I was referring to is the version number passed to the setup function using `version='X.X.X'`. In this case it is `0.1.0`.

Here's what I get when I change `use_scm_version={'write_to': 'package/VERSION'}` to `use_scm_version=True`:

``` shell
(testenv) C:\Dev\findig [develop +0 ~3 -0]> python setup.py bdist_wheel
running bdist_wheel
running build
running build_py
installing to build\bdist.win32\wheel
running install
running install_lib
creating build\bdist.win32\wheel
creating build\bdist.win32\wheel\findig
copying build\lib\findig\content.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
copying build\lib\findig\context.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
copying build\lib\findig\data_model.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
copying build\lib\findig\dispatcher.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
creating build\bdist.win32\wheel\findig\extras
copying build\lib\findig\extras\redis.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
\extras
copying build\lib\findig\extras\sql.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig\e
xtras
copying build\lib\findig\extras\__init__.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\fin
dig\extras
copying build\lib\findig\json.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
copying build\lib\findig\resource.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
creating build\bdist.win32\wheel\findig\tools
copying build\lib\findig\tools\counter.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findi
g\tools
copying build\lib\findig\tools\dataset.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findi
g\tools
creating build\bdist.win32\wheel\findig\tools\protector
copying build\lib\findig\tools\protector\scopeutil.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.d
ata\..\findig\tools\protector
copying build\lib\findig\tools\protector\__init__.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.da
ta\..\findig\tools\protector
copying build\lib\findig\tools\validator.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\fin
dig\tools
copying build\lib\findig\tools\__init__.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\find
ig\tools
copying build\lib\findig\utils.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
copying build\lib\findig\VERSION -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
copying build\lib\findig\wrappers.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
copying build\lib\findig\__init__.py -> build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\findig
running install_egg_info
running egg_info
writing requirements to Findig.egg-info\requires.txt
writing dependency_links to Findig.egg-info\dependency_links.txt
writing Findig.egg-info\PKG-INFO
writing top-level names to Findig.egg-info\top_level.txt
reading manifest template 'MANIFEST.in'
writing manifest file 'Findig.egg-info\SOURCES.txt'
Copying Findig.egg-info to build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.data\..\Findig-0.1.1.dev1+ng211
f867.d20150719-py3.4.egg-info
running install_scripts
creating build\bdist.win32\wheel\Findig-0.1.1.dev1+ng211f867.d20150719.dist-info\WHEEL
(testenv) C:\Dev\findig [develop +0 ~3 -0]>
```

**Comment by mr-rodgers** [comment]
Using the fail-fast branch, rather than the master branch, this is what I get:

```
Traceback (most recent call last):
  File "setup.py", line 52, in <module>
    'sql': ['sqlalchemy'],
  File "C:\Python34\lib\distutils\core.py", line 108, in setup
    _setup_distribution = dist = klass(attrs)
  File "C:\Dev\testenv\lib\site-packages\setuptools\dist.py", line 272, in __init__
    _Distribution.__init__(self,attrs)
  File "C:\Python34\lib\distutils\dist.py", line 280, in __init__
    self.finalize_options()
  File "C:\Dev\testenv\lib\site-packages\setuptools\dist.py", line 327, in finalize_options
    ep.load()(self, ep.name, value)
  File "c:\dev\findig\setuptools_scm-1.5.6.dev3+ng389811c-py3.4.egg\setuptools_scm\integration.py", line 17, in version_keyword
  File "c:\dev\findig\setuptools_scm-1.5.6.dev3+ng389811c-py3.4.egg\setuptools_scm\__init__.py", line 60, in get_version

  File "c:\dev\findig\setuptools_scm-1.5.6.dev3+ng389811c-py3.4.egg\setuptools_scm\__init__.py", line 39, in dump_version
TypeError: Can't convert 'tuple' object to str implicitly
```

Looking at the source code for the function that fails, it looks as though the write_to parameter expects either a ".txt" or ".py" file extension, and will fail otherwise. Changing my write_to file from `VERSION` to `VERSION.txt` does  resolve this, although forcing the .txt extension seems like an odd imposition.

**Comment by RonnyPfannschmidt** [comment]
the extension is used to decide how to export the data,

in hindight the approach was a unfortunate and bad idea

**Comment by RonnyPfannschmidt** [comment]
the main problem was solved in the fail-fast branch, the rest will be solved later in #36

thanks for bringing it to attention and for providing refinements to the fail-fast solution
